src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,22 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Int }
+
+-- | Similar to `sampleOn_`, except that the returned `Event` is a `Tuple` of
+-- | the sampled value _as well as_ the time since that value was pushed to the
+-- | stream (in milliseconds).
+withDelay :: forall a b. Event a -> Event b -> Event (Tuple Int a)
```

Comment:
I'm not against adding this, but it seems quite specific. What are you using it for? Can you please add an example to the docstring?